pkg/network/namescheme/networknamescheme.go

```diff
@@ -94,3 +97,34 @@ func generateOrdinalInterfaceName(idx int) string {
 	const ordinalIfacePrefix = "net"
 	return fmt.Sprintf("%s%d", ordinalIfacePrefix, idx)
 }
+
+// CreateNetworkNameSchemeByPodNetworkStatus create pod network name scheme according to the given VMI spec networks
+// and pod network status.
+// In case the pod network status has at least one interface with ordinal interface name it returns the ordinal network
+// name-scheme, Otherwise, returns the hashed network name scheme.
+func CreateNetworkNameSchemeByPodNetworkStatus(networks []v1.Network, podNetworkStatus map[string]networkv1.NetworkStatus) map[string]string {
+	if podHasOrdinalInterfaceName(podNetworkStatus) {
+		return CreateOrdinalNetworkNameScheme(networks)
+	}
+
+	return CreateNetworkNameScheme(networks)
+}
+
+// podHasOrdinalInterfaceName check if the given pod network status has at least one pod interface with ordinal name
+func podHasOrdinalInterfaceName(podNetworkStatus map[string]networkv1.NetworkStatus) bool {
+	for _, networkStatus := range podNetworkStatus {
+		if ordinalSecondaryInterfaceName(networkStatus.Interface) {
+			return true
+		}
+	}
+	return false
+}
+
+// ordinalSecondaryInterfaceName check if the given name is in form of the ordinal
+// name scheme (e.g.: net1, net2..).
+// Primary iface name (eth0) is treated as non-ordinal interface name.
+func ordinalSecondaryInterfaceName(name string) bool {
```

Comment:
please prefix the func name with `is` - maybe `isSecondaryInterfaceNameOrdinal` ?

Reply:
I prefer to keep the current naming, it reads batter IMO: (if ordinal-secondary-iface-name vs. if is ordinal-secondary-iface-name).

Reply:
prefixing functions that return booleans w/ `is` / `does` / ... is a good practice.

I won't insist further though.